src/Repositories/ContentRepository.php

```diff
@@ -1364,35 +1394,81 @@ public function retrieveFilter()
                     ->groupByField($this->groupByFields);
             $query = $subQuery;
 
+            //Group by instructors
             if ($this->groupByFields['is_a_related_content']) {
                 $query =
                     $this->query()
                         ->selectPrimaryColumns()
                         ->addSelect('inner_content.lessons_grouped_by_field as lessons_grouped_by_field')
                         ->addSubJoinToQuery($subQuery)
                         ->directPaginate($this->page, $this->limit)
-                        ->orderBy('slug', 'asc');
+                        ->orderBy('slug', $this->orderDirection);
+                if (!empty($this->typesToInclude) && (count($this->typesToInclude) <= 3)) {
+                    $query->selectRaw(' "'.$this->typesToInclude[0].'" as content_type');
+                }
                 $contentRows = $query->getToArray();
                 $contentIds = $this->getGroupByContentIds($contentRows);
                 $dataLookup = $this->getContentLookupByIds($contentIds);
 
                 $contentRows = $this->contentCompiledColumnTransformer->transform(Arr::wrap($contentRows)) ?? [];
 
                 $contentRows = $this->contentCompiledColumnTransformer->transformLessons(
-                    $contentRows,
-                    $dataLookup
-                ) ?? [];
+                        $contentRows,
+                        $dataLookup
+                    ) ?? [];
 
                 return $contentRows;
             }
-            $contentRows = $query->getToArray();
+
+            if (!empty($this->typesToInclude)) {
+                $query->selectRaw(' "'.$this->typesToInclude[0].'" as content_type');
+            }
+
+            //Group by associated table(style/genre, bpm, essentials, creativity, lifestyle ...)
+            if (!empty($this->groupByFields['associated_table'])) {
+                $orderBy = ('m'.'.'.$this->groupByFields['associated_table']['column']);
+                $db = (new ContentQueryBuilder(
+                    $this->connection(),
+                    $this->connection()
+                        ->getQueryGrammar(),
+                    $this->connection()
+                        ->getPostProcessor()
+                ))->from($this->groupByFields['associated_table']['table'].' as m');
+                $db->selectRaw(
+                    'm.'.$this->groupByFields['associated_table']['column'].' as grouped_by_field,
+                    m.'.$this->groupByFields['associated_table']['column'].' as id,	
+	                COUNT( DISTINCT(lessons.caontent_idd)) AS lessonsCount,
+	                GROUP_CONCAT(lessons.caontent_idd) as lessons_grouped_by_field, '.
+                    ' "'.$this->groupByFields['associated_table']['column'].'" as type'
+                );
+                $rq =
+                    $query->selectRaw(' railcontent_content.id as caontent_idd')
```

Comment:
Just a typo on `caontent_idd`
```suggestion
	                COUNT( DISTINCT(lessons.content_id)) AS lessonsCount,
	                GROUP_CONCAT(lessons.content_id) as lessons_grouped_by_field, '.
                    ' "'.$this->groupByFields['associated_table']['column'].'" as type'
                );
                $rq =
                    $query->selectRaw(' railcontent_content.id as content_id')
```